### DISCUSSION TITLE
Wagmi core: Error: No wagmi client found. Ensure you have set up a client

### DISCUSSION THREAD

**Discussion by philmetzger** [wevm/wagmi#1501] [Q&A] [answered]
I am using wagmi/core like this to sign a message:

```
const signature = await signMessage({
      message,
    });
```

However i get the error:
`Uncaught (in promise) Error: No wagmi client found. Ensure you have set up a client: https://wagmi.sh/react/client`

I have the WagmiConfig wrapping my app. So shouldnt this work?

**Comment by tmm** [comment, marked as answer]
Can you post a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? Possible something is set up incorrectly and can't tell without a full picture.

**Reply by philmetzger** [reply]
```
const selectChains = isDevEnv ? [chain.polygonMumbai] : [chain.mainnet, chain.polygon];

const { chains, provider, webSocketProvider } = configureChains(selectChains, [
  infuraProvider({ apiKey: process.env.NEXT_PUBLIC_INFURA_ID }),
]);

const client = createClient({
  autoConnect: true,
  connectors: [
    new InjectedConnector({ chains }),
    new CoinbaseWalletConnector({ chains, options: { appName: 'appName' } }),
    new WalletConnectConnector({ chains, options: { qrcode: true } }),
  ],
  provider,
  webSocketProvider,
});
<WagmiConfig client={client}>{children}</WagmiConfig>
```

Then within children I have a component that when a button is clicked calls this function:
```
import { signMessage } from '@wagmi/core'
const handleClick = async () => {
    const signed = await signMessage({ message: "message" });
}
```

And that throws the error

**Reply by tmm** [reply]
Try importing `signMessage` from `wagmi/actions` instead. Generally you don't want to mix using `@wagmi/core` and `wagmi`.

Since you are using React, you might want to use `useSignMessage` instead:

```ts
import { useSignMessage } from 'wagmi'

const { signMessageAsync } = useSignMessage()

const handleClick = async () => {
    const signed = await signMessageAsync({ message: "message" });
}
```

**Reply by philmetzger** [reply]
`import { signMessage } from 'wagmi/actions';` did the tricks, thanks for the tip!
